prometheus/prometheus.libsonnet

```diff
@@ -41,7 +44,7 @@ local kausal = import 'ksonnet-util/kausal.libsonnet';
       }),
     ]
     + (
-      if std.objectHas(this, 'prometheusAlerts') && std.prune(this.prometheusAlerts) != {}
+      if std.prune(this.prometheusAlerts) != {}
```

Comment:
Shouldn't you prune while defining your local above (l38)? You prune here while checking for an empty definition but you don't while creating your yaml. Isn't that inconsistent? How often do these maps have empty members?

Reply:
No, empty members sometimes have meaning, I have fallen in this trap before. :grimacing:

Reply:
Why prune at all then?